teamfoundationcontrol.cpp

```diff
@@ -231,3 +249,36 @@ Core::ShellCommand *TeamFoundation::Internal::TeamFoundationControl::createIniti
     command->addJob(settings.binaryPath(), getArgs, -1);
     return command;
 }
+
+////////////////////////////////////////////////////////////////////////////////
+void TeamFoundationControl::DirectoryTree::Add(const QString& directory)
+{
+    for (int i = 0; i < m_entries.size(); ++i) {
+        // candidate shadowed?
+        if (IsParentOrSame(m_entries[i], directory)) {
+            return;
+        }
+
+        // candidate shadows?
+        if (IsParentOrSame(directory, m_entries[i])) {
+            m_entries[i] = m_entries[m_entries.size()-1];
+            m_entries.pop_back();
+            --i;
+        }
+    }
+
+    m_entries << directory;
+}
+
+bool TeamFoundationControl::DirectoryTree::IsParentOrSame(const QString& lsh, const QString& rhs)
+{
+    return rhs.startsWith(lsh, Qt::CaseInsensitive);

```

Comment:
Doesn't really matter, if one (1) has a trailing slash and the other doesn't (2) then (2) will be taken because it is a prefix of (1)